File: pset2/readability/readability.c
#include<stdio.h>
#include<string.h>
#include<stdlib.h>
#include<cs50.h>
#include <math.h>

int main (void)
{
    string text= get_string("Text: ");
    int i=0;
    int word=1;
    int letter=0;
    int sentence=0;
    int grade;
    float index;
    float L;
    float S;
    while (text[i] != '\0'){

        if (((int)text[i]>=65 && (int)text[i]<=90) || ((int)text[i]>=97 && (int)text[i]<=122)){
            letter++;
        }
        else if (text[i] == ' '){
            word++;
        }
        else if ( text[i]== '.' || text[i]== '?' || text[i]== '!'){
            sentence++;
        }
        i++;
    }
    L=((float)letter / (float)word)*100;
    S=((float)sentence / (float)word)*100;
    index= 0.0588 * L - 0.296 * S - 15.8;
    grade= (int)round(index);
    printf("Words-> %d\tLetters-> %d\tSentences-> %d\n", word, letter, sentence);
    printf("%f , %f , %f\n", L,S,index);
    if (grade<1){
        printf("Before Grade 1\n");
        return 0;
    }
    if (grade> 16){
        printf("Grade 16+\n");
        return 0;
    }
    printf("Grade %d\n",grade);
}

File: pset6:Python/readability/readability.py
import math
# Take user input 
text= input("Text: ")
letters =0
words =0
sentences= 0
for run in text:
    # print(run)
    if run == " ":
        words+=1
    elif  run == "!" or run == "?" or run == ".":
        sentences+=1
    else:
        letters+=1
        
words+=1
L= (letters / words)*100
S= (sentences / words)*100
index= 0.0588 * L - 0.296 * S - 15.8;
grade= math.floor(index)
if grade >16:
    print("Grade 16+\n")
if grade<1:
    print("Before Grade 1\n")
else:
    print("Grade {}".format(grade))